tests/examplePythonTypes.py

```diff
@@ -0,0 +1,132 @@
+    processing.
+
+    Parameters
+    ----------
+    summary : `dict`
+        Simple dictionary mapping key performance metrics to a scalar
+        result.
+    output : `dict`
+        Structured nested data.
+    data : `list`, optional
+        Arbitrary array data.
+    """
+
+    def __init__(self, summary, output, data=None):
+        self.summary = summary
+        self.output = output
+        self.data = data
+
+    def exportAsDict(self):
+        """Convert object contents to a single python dict."""
+        exportDict = {"summary": self.summary,
+                      "output": self.output}
+        if self.data is not None:
+            exportDict["data"] = list(self.data)
+        return exportDict
+
+    @classmethod
+    def makeFromDict(cls, exportDict):
+        """Create a new object from a dict that is compatible with that
+        created by `exportAsDict`.
+
+        Parameters
+        ----------
+        exportDict : `dict`
+            `dict` with keys "summary", "output", and (optionally) "data".
+
+        Returns
+        -------
+        newobject : `MetricsExample`
+            New `MetricsExample` object.
+        """
+        data = None
+        if "data" in exportDict:
+            data = exportDict["data"]
+        return cls(exportDict["summary"], exportDict["output"], data)
+
+
+class MetricsExampleC(MetricsExample):
+    """Version of `MetricsExample` that we treat as a composite.
+    """
+
+    @staticmethod
+    def assembleMetrics(exportDict):
+        """Construct an object from a dict of components.
+
+        Parameters
+        ----------
+        exportDict : `dict`
+            `dict` with keys "summary", "output", and (optionally) "data".
+
+        Returns
+        -------
+        metrics : `MetricsExample`
+            Newly constructed instance.
+        """
+        return MetricsExample.makeFromDict(exportDict)
+
+    def disassembleMetrics(self, storageClass):
+        """Disassemble a metrics object into components.
+
+        Parameters
+        ----------
+        storageClass : `StorageClass`
+            The storage class to be associated with the disassembly.
+
+        Returns
+        -------
+        components : `dict`
+            Individual components with keys matching the keys of the components
+            defined in `storageClass` and values being a tuple with the first
+            element the component and the second element the `StorageClass` of
+            the component.
+
+        Raises
+        ------
+        e : ValueError
+            A requested component can not be found.
+        """
+        requested = set(storageClass.components.keys())
+        components = {}
+        for c in list(requested):
+            # Remove from list
+            requested.remove(c)
+            sc = storageClass.components[c]
+            if hasattr(self, c):
+                components[c] = (getattr(self, c), sc)
```

Comment:
I wonder if we can't make this dissasembler more general? Seems like extracting components based on attributes is probably almost always what we want.

Reply:
Yes. `daf_persistence` has a generic disassembler that checks for standard get methods as well as properties.